File: C++Verifier/src/expressions/TrueExpr.cpp
#ifndef TRUEEXPR_H
#define TRUEEXPR_H

#include <vector>
#include "BoolExpr.cpp"
// #include "../utils/Packet.cpp"

using namespace std;

class TrueExpr: public BoolExpr {
public:
        using BoolExpr::BoolExpr;

        void* evaluate(shared_ptr<Packet> p) {
            // cout << "In True evaluateA" << endl;
            bool * ret = new bool;
            *ret = !negated;
            return static_cast<void *> (ret);
        }

        bool evaluate(shared_ptr<Packet> p,  vector<string> constraints, vector<string> locationList, vector<string> variableList, int currentState, vector<shared_ptr<ConstraintTreeNode>> rootConstraints)  {
            // cout << "In True evaluateB" << endl;
            return !negated;
        }

        string toString() {
            string output = (negated ? "!" : "");
            return output + "True";
        }
};

#endif

File: verification/src/main/java/org/msr/mnr/verification/expressions/TrueExpr.java
package org.msr.mnr.verification.expressions;

import java.util.ArrayList;
import java.util.List;

import org.msr.mnr.verification.dsfa.ConstraintTreeNode;
import org.msr.mnr.verification.utils.Packet;

public class TrueExpr extends BoolExpr {
    private static final long serialVersionUID = -5278681248335478957L;

    @Override
    public Boolean evaluate(Packet p, ArrayList<int[]> constraints, ArrayList<String> locationList,
            ArrayList<String> variableList, Integer currentState,
            List<ConstraintTreeNode> rootConstraints) {
        return !negated; // normally true
    }

    @Override
    public Boolean evaluate(Packet p) {
        return !negated; // normally true
    }
    
    @Override
    public String toString() {
        return (negated ? "!" : "") + "True";
    }
}